linode/design-language-system
# tech story: Update packages to address dependabot security concerns

## Description 📝
Updates (most) packages to their latest version in order to address the dependabot security concerns. Removes package-lock.json since that is old/not necessary with yarn.lock

## Major Changes 🔄
- updated packages to latest version, including react, eslint and eslint related packages
- updated node version to v18.18.0, so that we could upgrade to @typescript/eslint/-- to v7+ (v8.5.0)
  - I'd tried to update to v20.17.0 like cloud manager but that's leading to some errors when running yarn generate - will look into some more

## Testing
- make sure running `yarn generate` doesn't result in file differences
- make sure storybook still works as expected
- double check that nothing broke in the code
